src/FloatingWindows/FLWIntegration.cpp

```diff
@@ -692,15 +695,31 @@ void FindAndQuarantine (lua_State *L)
     oss_script_name << "FlyWithLua Info: Function Script Name From Stack " << ScriptName;
     flywithlua::logMsg(logToAll, oss_script_name.str());
 
-    oss_script_path_name << "FlyWithLua Info: Full Script Path Name " << flywithlua::scriptDir << "/" << ScriptName;
-    flywithlua::logMsg(logToAll, oss_script_path_name.str());
-
-    oss_quarantine_path_name << "FlyWithLua Info: Full Quarantine Path Name " << flywithlua::quarantineDir <<
-                                 ScriptName << std::endl;
-    flywithlua::logMsg(logToAll, oss_quarantine_path_name.str());
+    oss_script_path_name << flywithlua::scriptDir << "/" << ScriptName;
+    flywithlua::logMsg(logToAll, "FlyWithLua Info: Full Script Path Name " + oss_script_path_name.str());
+    std::string script_path_name = oss_script_path_name.str();
 
-    // flywithlua::ReadAllScriptFiles();
+    oss_quarantine_path_name << flywithlua::quarantineDir << ScriptName;
+    flywithlua::logMsg(logToAll, "FlyWithLua Info: Full Quarantine Path Name " + oss_quarantine_path_name.str());
+    std::string quarantine_path_name = oss_quarantine_path_name.str();
 
+    result = rename(script_path_name.c_str(), quarantine_path_name.c_str());
+    if (result == 0)
+    {
+        flywithlua::logMsg(logToDevCon,
+               ("FlyWithLua Info: Moved Bad Script to " + quarantine_path_name));
+    }
+    else
+    {
+        flywithlua::logMsg(logToDevCon,
+               ("FlyWithLua Info: Could not move bad script to " + quarantine_path_name));
+        // LuaIsRunning = false;
+        // Added this break to prevent CTD from flooding the Log.txt file.
+        // break;
+    }
+    XPLMSpeakString("Please reload Your Lua Script files.");
+    // flywithlua::LuaIsRunning = false;
+    // return flywithlua::ReadAllScriptFiles();
```

Comment:
If this is enabled we get a CTD and using gdb here is where it crashes.
`Thread 1 (Thread 0x7ffff7f25a40 (LWP 5244)):
#0  0x00007fff9e4b0853 in propagatemark () from /home/bill/X-Plane_11.31r1_light/Resources/plugins/FlyWithLua/64/lin.xpl
#1  0x00007fff9e4b11e8 in lj_gc_fullgc () from /home/bill/X-Plane_11.31r1_light/Resources/plugins/FlyWithLua/64/lin.xpl
#2  0x00007fff9e49d825 in lua_gc () from /home/bill/X-Plane_11.31r1_light/Resources/plugins/FlyWithLua/64/lin.xpl
#3  0x00007fff9e3e9b88 in flywithlua::ReadScriptFile(char const*) () from /home/bill/X-Plane_11.31r1_light/Resources/plugins/FlyWithLua/64/lin.xpl
#4  0x00007fff9e3f0491 in flywithlua::ReadAllScriptFiles() () from /home/bill/X-Plane_11.31r1_light/Resources/plugins/FlyWithLua/64/lin.xpl
#5  0x00007fff9e3f2201 in flywithlua::FlyWithLuaMenuHandler(void*, void*) () from /home/bill/X-Plane_11.31r1_light/Resources/plugins/FlyWithLua/64/lin.xpl
#6  0x00007fff9ea17f9f in XPLMHandleMenuItemPick(int) () from /home/bill/X-Plane_11.31r1_light/Resources/plugins/XPLM_64.so
#7  0x0000000001129182 in XPPHandleMenu(int) ()
#8  0x000000000083c7a3 in gui_menubar_widget::end_drag(float, float, gui_menubar_widget::end_mode) ()
#9  0x00000000008283ea in gui_event_dispatcher::event_mouse_up(click_event_info_t const&) ()
#10 0x000000000099f1ad in xplane_window::mouse_up(void*, int, int, int) ()
#11 0x00000000014d1538 in WIN_event_loop(int) ()
#12 0x00000000014d16c7 in WIN_run_app ()
#13 0x000000000049d65f in main ()`

Reply:
I think the problem is that  ReadAllScriptFiles() destroys Lua context. This means that you can't do this in a function, which was originally called by Lua: once that function returns, control is passed back to Lua is the *destroyed* context. From that point, anything can happen, including a crash.

A workaround is detect the bad script name and throw an exception. The exception will unwind Lua stack, and then we'll catch it at the top, where we originally passed control to Lua. we can create a custom exception class, which will carry the name of the bad script. At that point it should be safe to quarantine the script, re-initialize Lua environment and try again.

Reply:
I am using a exception that was already happening and then trying to reload the remaining scripts again. I need a little help in how to create a exception here if possible.